specs/interop.md

```diff
@@ -0,0 +1,502 @@
+<!-- START doctoc generated TOC please keep comment here to allow auto update -->
+<!-- DON'T EDIT THIS SECTION, INSTEAD RE-RUN doctoc TO UPDATE -->
+**Table of Contents**
+
+- [Interoperability](#interoperability)
+  - [Overview](#overview)
+  - [The Dependency Set](#the-dependency-set)
+    - [Chain ID](#chain-id)
+  - [Interop Network Upgrade](#interop-network-upgrade)
+  - [Derivation Pipeline](#derivation-pipeline)
+    - [Depositing a Relaying Message](#depositing-a-relaying-message)
+    - [Safety](#safety)
+  - [State Transition Function](#state-transition-function)
+  - [Predeploys](#predeploys)
+    - [CrossL2Outbox](#crossl2outbox)
+      - [Initiating Messages](#initiating-messages)
+    - [CrossL2Inbox](#crossl2inbox)
+      - [Invariants](#invariants)
+        - [Only EOA Invariant](#only-eoa-invariant)
+      - [Relaying Messages](#relaying-messages)
+    - [L2ToL2CrossDomainMessenger](#l2tol2crossdomainmessenger)
+      - [Transferring Ether in a Cross Chain Message](#transferring-ether-in-a-cross-chain-message)
+      - [Interfaces](#interfaces)
+        - [Sending Messages](#sending-messages)
+        - [Relaying Messages](#relaying-messages-1)
+    - [L1Block](#l1block)
+  - [SystemConfig](#systemconfig)
+    - [`DEPENDENCY_SET` UpdateType](#dependency_set-updatetype)
+  - [L1Attributes](#l1attributes)
+  - [Fault Proof](#fault-proof)
+  - [Sequencer Policy](#sequencer-policy)
+  - [Block Building](#block-building)
+  - [Sponsorship](#sponsorship)
+  - [Security Considerations](#security-considerations)
+    - [Forced Inclusion of Cross Chain Messages](#forced-inclusion-of-cross-chain-messages)
+    - [Cross Chain Message Latency](#cross-chain-message-latency)
+    - [Dynamic Size of L1 Attributes Transaction](#dynamic-size-of-l1-attributes-transaction)
+    - [Maximum Size of the Dependency Set](#maximum-size-of-the-dependency-set)
+    - [Mempool](#mempool)
+    - [Reliance on History](#reliance-on-history)
+
+<!-- END doctoc generated TOC please keep comment here to allow auto update -->
+
+# Interoperability
+
```

Comment:
Should add in the exact mapping to the `eth_getLogs` query from the event in the outbox

Reply:
pseudocode has been added